helix-zk/src/main/java/org/apache/helix/zkscale/zk/zookeeper/ZkClient.java

```diff
@@ -0,0 +1,1755 @@
+package org.apache.helix.zkscale.zk.zookeeper;
+
+
+import javax.management.JMException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.OptionalLong;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.TimeUnit;
+
+import org.I0Itec.zkclient.DataUpdater;
+import org.I0Itec.zkclient.ExceptionUtil;
+import org.I0Itec.zkclient.IZkChildListener;
+import org.I0Itec.zkclient.IZkConnection;
+import org.I0Itec.zkclient.IZkDataListener;
+import org.I0Itec.zkclient.IZkStateListener;
+import org.I0Itec.zkclient.ZkLock;
+import org.I0Itec.zkclient.exception.ZkBadVersionException;
+import org.I0Itec.zkclient.exception.ZkException;
+import org.I0Itec.zkclient.exception.ZkInterruptedException;
+import org.I0Itec.zkclient.exception.ZkNoNodeException;
+import org.I0Itec.zkclient.exception.ZkNodeExistsException;
+import org.I0Itec.zkclient.exception.ZkTimeoutException;
+import org.I0Itec.zkclient.serialize.ZkSerializer;
+//import org.apache.helix.HelixException;
+import org.apache.helix.zkscale.zk.ZkScaleException;
+//import org.apache.helix.ZNRecord;
+//import org.apache.helix.api.listeners.PreFetch;
+import org.apache.helix.zkscale.zk.PreFetch;
+import org.apache.helix.zkscale.zk.BasicZkSerializer;
+import org.apache.helix.zkscale.zk.PathBasedZkSerializer;
+import org.apache.helix.zkscale.zk.ZkAsyncCallbacks;
+import org.apache.helix.zkscale.zk.zookeeper.ZkEventThread.ZkEvent;
+import org.apache.helix.zkscale.zk.util.ZkClientMonitor;
+import org.apache.helix.zkscale.zk.util.ExponentialBackoffStrategy;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.KeeperException.ConnectionLossException;
+import org.apache.zookeeper.KeeperException.SessionExpiredException;
+import org.apache.zookeeper.Op;
+import org.apache.zookeeper.OpResult;
+import org.apache.zookeeper.WatchedEvent;
+import org.apache.zookeeper.Watcher;
+import org.apache.zookeeper.Watcher.Event.EventType;
+import org.apache.zookeeper.Watcher.Event.KeeperState;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.ZooKeeper;
+import org.apache.zookeeper.data.ACL;
+import org.apache.zookeeper.data.Stat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstracts the interaction with zookeeper and allows permanent (not just one time) watches on
+ * nodes in ZooKeeper.
+ * WARN: Do not use this class directly, use {@link org.apache.helix.manager.zk.ZkClient} instead.
+ */
+public class ZkClient implements Watcher {
```

Comment:
We could technically remove this class in the helix-core module because users shouldn't be using this directly anyways?